src/app/authentication/logout.ts

```diff
@@ -1,15 +1,27 @@
+import { removeDeviceTokens } from "@app/users/remove-device-tokens"
 import { AuthWithPhonePasswordlessService, MissingSessionIdError } from "@services/kratos"
 
-export const logoutToken = async (
-  sessionId: SessionId | undefined,
-): Promise<boolean | ApplicationError> => {
+export const logoutToken = async ({
+  userId,
+  deviceToken,
+  sessionId,
+}: {
+  userId: UserId
+  deviceToken: DeviceToken | undefined
+  sessionId: SessionId | undefined
+}): Promise<boolean | ApplicationError> => {
   if (!sessionId) return new MissingSessionIdError()
 
   const authService = AuthWithPhonePasswordlessService()
   const kratosResult = await authService.logoutToken({ sessionId })
   if (kratosResult instanceof Error) {
     return kratosResult
   }
+
+  if (deviceToken) {
+    await removeDeviceTokens({ userId, deviceTokens: [deviceToken] })
```

Comment:
dont we need to return the error?

Reply:
don't think that's useful or necessary. main action of logout would be to deactivate the session.